src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+
 public class Duke {
-    public static void main(String[] args) {
+    public static ArrayList<Task> manager = new ArrayList<>();
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        greet();
+        addLineBreak();
+
+        while (true) {
+            String instruct = br.readLine();
+            try {
+                if (instruct.equals("bye")) {
+                    endSession();
+                    break;
+                } else if (instruct.equals("list")) {
+                    reportList();
+                } else {
+                    String[] details = instruct.split(" ");
+                    if (details[0].equals("mark")) {
+                        markAsDone(Integer.parseInt(details[1]));
+                    } else if (details[0].equals("unmark")) {
+                        markNotDone(Integer.parseInt(details[1]));
+                    } else if (details[0].equals("delete")) {
+                        deleteTask(Integer.parseInt(details[1]));
+                    } else {
+                        String taskType = details[0];
+                        if (!(taskType.equals("todo") || taskType.equals("deadline")
+                                || taskType.equals("event"))) {
+                            throw new CustomException("sorry, this isn't a valid command yet!");
+                        } else {
+                            addTask(taskType, instruct);
+                        }
```

Comment:
Maybe using a switch case for the different commands might make the code easier to read?